traitlets/traitlets.py

```diff
@@ -1419,8 +1431,10 @@ def validate(self, obj, value):
 class Enum(TraitType):
     """An enum that whose value must be in a given sequence."""
 
-    def __init__(self, values, default_value=None, **metadata):
+    def __init__(self, values, default_value=NoDefaultSpecified, **metadata):
         self.values = values
+        if metadata.get('allow_none', False) and default_value is NoDefaultSpecified:
+            default_value = None
```

Comment:
This makes the default value depend on the value of `allow_none`. Why not `Undefined`?